### ISSUE TITLE
MariaDB Connector/J 3.0 - Binary Data Formatter Hex broken

### ISSUE THREAD

**Issue by aa-matthias** [dbeaver/dbeaver#16847]
#### System information: 
- Operating system (distribution) and version (Windows 11 21H2)
- DBeaver version 22.1.0
- Additional extensions 

#### Connection specification:
- Database name and version MariaDB 10.9.1
- Driver name MariaDB Connector/J 3.0 (3.0.3, 3.0.4, 3.0.5)
- Do you use tunnels or proxies (SSH, SOCKS, etc)? SSH

#### Describe the problem you're observing:
Setting Binary Format to Hex has no effect and data is not displayed correctly when using MariaDB Connector/J 3.0 - it's working fine with MariaDB Connector/J 2.0 (2.7.5)

#### Steps to reproduce, if exist:
Set Binary Display Format to Hex
Set Up a new Connection and specify any 3.*.* release from MariaDB Connector/J in the library config

**Comment by LonwoLonwo** [comment]
Hello @aa-matthias 

For now, I can't reproduce your issue.

Can you please provide a table DDL and sample data?
A video of the issue will also be beneficial.

**Comment by aa-matthias** [comment]
Hi @LonwoLonwo 

```sql
            CREATE TABLE IF NOT EXISTS `flow` (
                `id`                    BINARY(16)      NOT NULL,
                `name`                  VARCHAR(255)    COLLATE utf8mb4_unicode_ci  NOT NULL,
                `description`           MEDIUMTEXT      COLLATE utf8mb4_unicode_ci  NULL,
                `event_name`            VARCHAR(255)    NOT NULL,
                `priority`              INT(11)         NOT NULL DEFAULT 1,
                `payload`               LONGBLOB        NULL,
                `invalid`               TINYINT(1)      NOT NULL DEFAULT 0,
                `active`                TINYINT(1)      NOT NULL DEFAULT 0,
                `custom_fields`         JSON            NULL,
                `created_at`            DATETIME(3)     NOT NULL,
                `updated_at`            DATETIME(3)     NULL,
                PRIMARY KEY (`id`),
                INDEX `idx.flow.event_name` (`event_name`, `priority`),
                CONSTRAINT `json.flow.custom_fields` CHECK (JSON_VALID(`custom_fields`))
            ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;

INSERT INTO flow
(id, name, description, event_name, priority, payload, invalid, active, custom_fields, created_at, updated_at)
VALUES(0x03533CC8D7E54402BA0750AEFE5F560A, 'Payment enters status paid', NULL, 'state_enter.order_transaction.state.paid', 1, NULL, 0, 0, NULL, '2022-03-09 13:20:32.230', '2022-05-18 05:21:37.569');
```

![image](https://user-images.githubusercontent.com/84960805/175876386-298cfff3-1c34-4a2d-a6b7-fb0fdd9010c9.png)

**Comment by tgross35** [comment]
How does dbeaver determine what driver JDBC driver to install by default? 2.7.1 is what the latest update installed for me, but that's from 2020. 3.0.5 is the latest https://mariadb.com/kb/en/mariadb-connector-j-releases/

**Comment by aa-matthias** [comment]
@emironovDB - for me it's solved by updating to version 3.0.7 - the changelog mentions couple of changes related to binary display

https://github.com/mariadb-corporation/mariadb-connector-j/releases/tag/3.0.7

**Comment by LonwoLonwo** [comment]
THanks for investigation everyone.

Probably we need to update default driver version for MariaDB.
